nvm.sh

```diff
@@ -2249,6 +2249,46 @@ nvm() {
       nvm_err "\$NPM_CONFIG_PREFIX: '$(nvm_sanitize_path "$NPM_CONFIG_PREFIX")'"
       nvm_err "\$NVM_NODEJS_ORG_MIRROR: '${NVM_NODEJS_ORG_MIRROR}'"
       nvm_err "\$NVM_IOJS_ORG_MIRROR: '${NVM_IOJS_ORG_MIRROR}'"
+      nvm_err "shell version: '$(${SHELL} --version | command head -n 1)'"
+      nvm_err "uname -a: '$(uname -a | awk '{$2=""; print}' | xargs)'"
+      if [ "$(nvm_get_os)" = "darwin" ] && nvm_has sw_vers; then
+        nvm_err "OS version: $(sw_vers | command awk '{print $2}' | command xargs)"
+      elif [ -r "/etc/issue" ]; then
+        nvm_err "OS version: $(command head -n 1 /etc/issue | command sed 's/\\.//g')"
+      fi
+      if nvm_has "curl"; then
+        local CURL
+        if type curl | command grep -q hashed; then
+          CURL="$(type curl | command sed -E 's/\(|)//g' | command awk '{print $4}')"
```

Comment:
it seems like this logic is repeated three times; can we consolidate it into a `nvm_command_info` function? I'm thinking `CURL="$(nvm_command_info curl)"`